trulens_eval/trulens_eval/ux/components.py

```diff
@@ -157,6 +158,14 @@ def draw_llm_info(query: JSONPath, component: ComponentView) -> None:
                 """
     df = pd.DataFrame.from_dict(llm_kv, orient='index').transpose()
 
+    # Redact any column whose name indicates it might be a secret.
+    for col in df.columns:
+        if should_redact_key(col):
```

Comment:
Do we need to do this on DB read of previously ingested keys? Or do we want to migrate the DB to remove them permanently?

Reply:
This is a point of discussion, should we redact before adding to database or only before showing to the user. There are some implications of the first one for my big replay task. Will schedule a time to discuss but can we go with the second option for now?

Reply:
yes lets do at least one, good with either for short term. I'm not 100% on which one is "second option" so if you can comment back explicitly on which one you choose.

Reply:
Redact when presenting to the user but keep the keys in database (so that they can be used to replay chains).